packages/web-app-files/src/uppyUploadPlugin.ts

```diff
@@ -0,0 +1,136 @@
+import Uppy from '@uppy/core'
+import BasePlugin from '@uppy/core/lib/BasePlugin.js'
+import { dirname } from 'path'
+import * as uuid from 'uuid'
+import { Ref, unref } from 'vue'
+import { Store } from 'vuex'
+import { RouteLocationNormalizedLoaded } from 'vue-router'
+import { Resource, SpaceResource } from 'web-client/src'
+import { urlJoin } from 'web-client/src/utils'
+import { UppyResource } from 'web-runtime/src/composables/upload'
+import { UppyService } from 'web-runtime/src/services/uppyService'
+
+/**
+ * Prepares all input files and adds necessary data for the upload. This is only used for
+ * the cloud import at the moment.
+ *
+ * It basically does the same as the inputFilesToUppyFiles method from the upload helpers.
+ * TODO: Remove the upload helpers and use this plugin for upload handling in general.
+ */
+export class HandleUpload extends BasePlugin {
+  route: Ref<RouteLocationNormalizedLoaded>
+  store: Store<any>
+  uppyService: UppyService
+
+  constructor(uppy, opts) {
+    super(uppy, opts)
+    ;(this as any).id = opts.id || 'HandleUpload'
+    ;(this as any).type = 'modifier'
+    this.route = opts.route
+    this.store = opts.store
+    this.uppyService = opts.uppyService
+
+    this.prepareUpload = this.prepareUpload.bind(this)
+  }
+
+  get _uppy(): Uppy {
+    return (this as any).uppy
+  }
+
+  get currentFolder(): Resource {
+    return this.store.getters['Files/currentFolder']
+  }
+
+  get space(): SpaceResource {
+    return this.store.getters['runtime/spaces/spaces'].find(
+      ({ id }) => id === this.currentFolder.storageId
+    )
+  }
+
+  /**
+   * Get the relative path of the file when the file was inside a directory on the client computer.
+   * @param file
+   */
+  getRelativeFilePath = (file): string | undefined => {
+    const relativePath = file.webkitRelativePath || (file as any).relativePath
```

Comment:
types could be better ;-)

Reply:
Totally agree! I'll tackle this in the follow up PR because it refactors the plugin by quite a lot.